### PR TITLE
Add new test consumer that returns the given error

### PR DESCRIPTION
**Pull request by bogdandrutu**
Updates #2477

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: consumer/consumertest/err.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+package consumertest
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+type errConsumer struct {
```
Comment (tigrannajaryan) [inline consumer/consumertest/err.go:24]: Not: maybe name it `failerConsumer`?

File: consumer/consumertest/err.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+func (er *errConsumer) ConsumeMetrics(context.Context, pdata.Metrics) error {
+	return er.err
+}
+
+func (er *errConsumer) ConsumeLogs(context.Context, pdata.Logs) error {
+	return er.err
+}
+
+// NewTracesErr returns a consumer.TracesConsumer that just drops all received data and returns the given error.
+func NewTracesErr(err error) consumer.TracesConsumer {
```
Comment (tigrannajaryan) [inline consumer/consumertest/err.go:41]: Nit: Maybe `NewTracesFailer`.

**Review by tigrannajaryan — APPROVED** [review]

File: consumer/consumertest/err.go (hunk 1 above)
Comment (bogdandrutu) [reply consumer/consumertest/err.go:24]: Not sure if that is a word...

Comment (tigrannajaryan) [reply consumer/consumertest/err.go:24]: Almost :-) https://en.wiktionary.org/wiki/failer

>One who fails.

**Comment by tigrannajaryan** [comment]
Feel free to merge.

**Pull request merged**